code/ai/aigoals.cpp

```diff
@@ -1419,7 +1459,8 @@ int ai_mission_goal_achievable( int objnum, ai_goal *aigp )
 	//  these orders are always achievable.
 	if ( (aigp->ai_mode == AI_GOAL_KEEP_SAFE_DISTANCE)
 		|| (aigp->ai_mode == AI_GOAL_CHASE_ANY) || (aigp->ai_mode == AI_GOAL_STAY_STILL)
-		|| (aigp->ai_mode == AI_GOAL_PLAY_DEAD) || (aigp->ai_mode == AI_GOAL_PLAY_DEAD_PERSISTENT) )
+		|| (aigp->ai_mode == AI_GOAL_PLAY_DEAD) || (aigp->ai_mode == AI_GOAL_PLAY_DEAD_PERSISTENT) 
+		|| (aigp->ai_mode == AI_GOAL_LUA))
```

Comment:
Does it make sense to delegate the achievable check to a scripting function?

Reply:
It might make sense to allow the modder to define a "is AI valid for current situation" function. Though, I'd certainly make it optional, and defer to always assuming it's achievable otherwise. In that case, it probably should be put into a future PR?

Reply:
I think the first "achievable" goal will always be executed.  So the question here is whether the modder will always want to run Lua code, as opposed to deferring it for some period to run standard behaviors like attacking.  But doing that in a future PR sounds fine.